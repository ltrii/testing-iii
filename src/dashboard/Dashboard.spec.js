// Test away
import React from 'react';
import Dashboard from './Dashboard';
import { cleanup, fireEvent, render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';

afterEach(cleanup);

describe('Dashboard component', () => {
    
    it("Contains necesscary components", () => {
        const component = render(<Dashboard />);
        component.getByText(/open/i);
        component.getByText(/unlocked/i);
      });
    test('Closing Works', () => {
        const button1 = render(<Dashboard />).getByTestId("clopenbtn");
        const dashboard = render(<Dashboard />);
        fireEvent.click(button1);
        const lockhold = dashboard.getByTestId("clopenbox");
        expect(lockhold).toHaveTextContent('Closed');
    })
    test('Locking Works', () => {
        const button1 = render(<Dashboard />).getByTestId("clopenbtn");
        const button2 = render(<Dashboard />).getByTestId("lockbtn");
        const dashboard = render(<Dashboard />);
        fireEvent.click(button1);
        fireEvent.click(button2);
        const lockhold = dashboard.getByTestId("lockbox");
        expect(lockhold).toHaveTextContent('Locked');
    })
    test('Unlocking Works', () => {
        const button1 = render(<Dashboard />).getByTestId("clopenbtn");
        const button2 = render(<Dashboard />).getByTestId("lockbtn");
        const dashboard = render(<Dashboard />);
        fireEvent.click(button1);
        fireEvent.click(button2);
        fireEvent.click(button2);
        fireEvent.click(button1);
        const lockhold = dashboard.getByTestId("lockbox");
        expect(lockhold).toHaveTextContent('Unlocked');
    })
})