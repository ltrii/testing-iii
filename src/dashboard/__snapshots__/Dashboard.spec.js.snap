// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Snapshot test 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="lockbox"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="clopenbox"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lockbtn"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="clopenbtn"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
